test/e2e/performanceprofile/functests/10_performance_ppc/ppc.go

```diff
@@ -0,0 +1,200 @@
+		It("[test_id:41405] Verify PPC script fails when the splitting of reserved cpus and single numa-node policy is specified", func() {
+			defaultArgs := []string{
+				"run",
+				"--entrypoint",
+				"performance-profile-creator",
+				"-v",
+			}
+			cmdArgs := []string{
+				fmt.Sprintf("%s:%s:z", mustgatherDir, mustgatherDir),
+				ntoImage,
+				"--mcp-name=worker",
+				fmt.Sprintf("--reserved-cpu-count=%d", 2),
+				fmt.Sprintf("--rt-kernel=%t", true),
+				fmt.Sprintf("--power-consumption-mode=%s", "low-latency"),
+				fmt.Sprintf("--split-reserved-cpus-across-numa=%t", true),
+				fmt.Sprintf("--topology-manager-policy=%s", "single-numa-node"),
+				fmt.Sprintf("--must-gather-dir-path=%s", mustgatherDir),
+			}
+			podmanArgs := []string{}
+			podmanArgs = append(defaultArgs, cmdArgs...)
+			session, err := ppcIntgTest.PodmanAsUserBase(podmanArgs, false, false)
+			Expect(err).ToNot(HaveOccurred(), "Podman command failed")
+			output := session.Wait(20).Err.Contents()
+			errString := "Error: failed to obtain data from flags not appropriate to split reserved CPUs in case of topology-manager-policy: single-numa-node"
+			ok, err := regexp.MatchString(errString, string(output))
+			if ok {
+				testlog.Info(errString)
+			}
+			Eventually(session).Should(gexec.Exit(1))
+		})
+
+		It("[test_id:41419] Verify PPC script fails when reserved cpu count is 2 and requires to split across numa nodes", func() {
+			defaultArgs := []string{
+				"run",
+				"--entrypoint",
+				"performance-profile-creator",
+				"-v",
+			}
+			cmdArgs := []string{
+				fmt.Sprintf("%s:%s:z", mustgatherDir, mustgatherDir),
+				ntoImage,
+				"--mcp-name=worker",
+				fmt.Sprintf("--reserved-cpu-count=%d", 2),
+				fmt.Sprintf("--rt-kernel=%t", true),
+				fmt.Sprintf("--power-consumption-mode=%s", "low-latency"),
+				fmt.Sprintf("--split-reserved-cpus-across-numa=%t", true),
+				fmt.Sprintf("--must-gather-dir-path=%s", mustgatherDir),
+			}
+			podmanArgs := []string{}
+			podmanArgs = append(defaultArgs, cmdArgs...)
+			session, err := ppcIntgTest.PodmanAsUserBase(podmanArgs, false, false)
+			Expect(err).ToNot(HaveOccurred(), "Podman command failed")
+			output := session.Wait(20).Err.Contents()
+			errString := "Error: failed to compute the reserved and isolated CPUs: can't allocate odd number of CPUs from a NUMA Node"
+			ok, err := regexp.MatchString(errString, string(output))
+			Expect(err).ToNot(HaveOccurred(), "did not fail with Expected:%s failure", errString)
+			if ok {
+				testlog.Info(errString)
+			}
+			Eventually(session).Should(gexec.Exit(1))
+		})
+
+		It("[test_id:41420] Verify PPC script fails when reserved cpu count is more than available cpus", func() {
+			defaultArgs := []string{
+				"run",
+				"--entrypoint",
+				"performance-profile-creator",
+				"-v",
+			}
+			cmdArgs := []string{
+				fmt.Sprintf("%s:%s:z", mustgatherDir, mustgatherDir),
+				ntoImage,
+				"--mcp-name=worker",
+				fmt.Sprintf("--reserved-cpu-count=%d", 100),
```

Comment:
I know we don't usually have systems with CPU amount greater than 80, but it would be more certain to set this filed based on the actual CPU count on the system that runs this test, for example get the CPU count and add 10

Reply:
Majority of test is reading mustgather directory and uses that to generate Profile, the only time we use the cluster is to verify if the generated Performance Profile  is valid, So it doesn't matter where this test is executed. I want to make the  tests agnostic to cluster and only required is mustgather and a connection to cluster to verify if generated PP is valid. 

I can increase the number to another 100 or so

Reply:
got it, thanks for clarifying